WuV/typing.tex

```diff
@@ -367,6 +371,22 @@ \subsection{Rigorous Definition}
 }
 \end{lstlisting}
 The details can vary, and special care must be taken in programming languages where initialization may have side effects.
+Methods of OO-languages arise as special case of fields whose types are the method signatures and whose definitions are the method bodies.

```

Comment:
Below, the notes contain a discussion on the difficulties of defining the mixin wrt. mutual recursion. Hence, introduce an intuition of how methods can be represented via ADTs *here* to later be able to talk about mutual recursion intuitively. Particularly for students without background in formalizing things the MMT way.